ang/test/civicase/DashboardTab.spec.js

```diff
@@ -208,12 +180,282 @@
+          });
+
+          it('is set to "myActivities" by default', function () {
+            expect($scope.newMilestonesPanel.custom.involvementFilter).toEqual({
+              '@involvingContact': 'myActivities'
+            });
+          });
+
+          describe('when it changes', function () {
+            beforeEach(function () {
+              spyOn($rootScope, '$broadcast');
+
+              $scope.newMilestonesPanel.custom.involvementFilter = { '@involvingContact': '' };
+              $scope.$digest();
+            });
+
+            it('broadcasts a "civicaseActivityFeed.query" event', function () {
+              expect($rootScope.$broadcast).toHaveBeenCalledWith(
+                'civicaseActivityFeed.query',
+                $scope.newMilestonesPanel.custom.involvementFilter,
+                $scope.newMilestonesPanel.query.params,
+                true
+              );
+            });
+          });
+        });
+      });
+
+      describe('when the relationship type changes', function () {
+        var newFilterValue, newType;
+
+        beforeEach(function () {
+          newType = 'is_involed';
+          newFilterValue = 'bar';
+
+          $scope.filters.caseRelationshipType = newType;
+          $scope.activityFilters.case_filter.foo = newFilterValue;
+
+          $scope.$digest();
+        });
+
+        it('adds the properties of the `case_filter` object to the query params', function () {
+          expect($scope.newMilestonesPanel.query.params.case_filter.foo).toEqual(newFilterValue);
+        });
+      });
+    });
+
+    /**
+     * Initializes the controller and digests the scope
+     */
     function initController () {
       $controller('dashboardTabController', {
         $scope: $scope,
+        formatActivity: formatActivityMock,
         formatCase: formatCaseMock
       });
       $scope.$digest();
     }
+
+    /**
+     * Given a list of mocked results, it will find all the contact ids in the
+     * given "contacts list" property and returns the total of all the ids and of
+     * all the unique ids
+     *
+     * @param {Array} mockResults
+     * @param {Array} contactsListKey
+     * @return {Object}
+     */
+    function countTotalAndUniqueContactIds (mockedResults, contactsListKey) {
+      var total, uniq;
+
+      uniq = _(mockedResults)
+        .pluck(contactsListKey).flatten()
+        .pluck('contact_id')
+        .tap(function (nonUniq) {
+          total = nonUniq;
+          return nonUniq;
+        })
+        .uniq().value();
+
+      return { total: total, uniq: uniq };
+    }
+
+    /**
+     * Returns the start and end of the given range (week/month) formatted
+     * in the given format
+     *
+     * @param {String} range
+     * @param {String} format
+     * @return {Array}
+     */
+    function getStartEndOfRange (range, format, useNowAsStart) {
+      var now = moment();
+
+      return [
+        (useNowAsStart ? now : now.startOf(range)).format(format),
```

Comment:
can we refactor this bit here? I rather have this in a different line than mix it with all these brackets, parenthesis, etc.

Maybe: